### PR TITLE
[WIP] Example - this demonstrates code integrating with Spring Data R2DBC

### PR DESCRIPTION
**Pull request by dzou**
This is an example of using the Spanner R2DBC driver with Spring Data r2dbc.

Context: #84.

### CODE REVIEW

File: src/test/java/com/google/cloud/spanner/r2dbc/it/SpringDataIT.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  @Test
+  public void tryReadQuery() {
+    List<String> rows = databaseClient
+        .select()
+        .from("POKEMONS")
+        .project("name", "level")
+        .orderBy(Sort.by(Order.desc("name")))
+        .map((row, rowMetadata) -> row.get(0, String.class))
```
Comment (meltsufin) [inline src/test/java/com/google/cloud/spanner/r2dbc/it/SpringDataIT.java]: Can you try the `.as(Pokemon.class)` to see if it can do some basic ORM?

**Comment by meltsufin** [comment]
See: https://github.com/spring-projects/spring-data-r2dbc#examples-using-reactive-repositories

File: src/test/java/com/google/cloud/spanner/r2dbc/it/SpringDataIT.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        .all()
+        .collectList()
+        .block();
+
+    System.out.println(rows);
+    assertThat(rows).containsExactlyInAnyOrder("Pokachu", "Scyther");
```
Comment (ChengyuanZhao) [inline src/test/java/com/google/cloud/spanner/r2dbc/it/SpringDataIT.java]: Pokachu

Comment (elefeint) [reply src/test/java/com/google/cloud/spanner/r2dbc/it/SpringDataIT.java]: A rarely seen Russian variety. ("Poka" means "bye" in Russian).

File: src/test/java/com/google/cloud/spanner/r2dbc/it/SpannerR2dbcDialect.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  private static final LimitClause LIMIT_CLAUSE = new LimitClause() {
+    @Override
+    public String getLimit(long limit) {
+      return "LIMIT " + limit;
+    }
+
+    @Override
+    public String getOffset(long offset) {
+      return "LIMIT 999999999 OFFSET " + offset;
```
Comment (elefeint) [inline src/test/java/com/google/cloud/spanner/r2dbc/it/SpannerR2dbcDialect.java]: LIMIT is not optional?

File: src/test/java/com/google/cloud/spanner/r2dbc/it/SpannerR2dbcDialectProvider.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import io.r2dbc.spi.ConnectionFactory;
+import java.util.Optional;
+import org.springframework.data.r2dbc.dialect.DialectResolver.R2dbcDialectProvider;
+import org.springframework.data.r2dbc.dialect.R2dbcDialect;
+
+public class SpannerR2dbcDialectProvider implements R2dbcDialectProvider {
+
+  @Override
+  public Optional<R2dbcDialect> getDialect(ConnectionFactory connectionFactory) {
+    if (connectionFactory.getMetadata().getName().equals("Cloud Spanner")) {
```
Comment (elefeint) [inline src/test/java/com/google/cloud/spanner/r2dbc/it/SpannerR2dbcDialectProvider.java]: Hmm, now I wonder if we should have made SpannerConnectionFactoryProvider.DRIVER_NAME be "Cloud Spanner".

File: src/test/java/com/google/cloud/spanner/r2dbc/it/SpringDataIT.java (hunk 2 above)
Comment (meltsufin) [reply src/test/java/com/google/cloud/spanner/r2dbc/it/SpringDataIT.java]: Nice play on the name to avoid copyright claims. :stuck_out_tongue_winking_eye:

Comment (ChengyuanZhao) [reply src/test/java/com/google/cloud/spanner/r2dbc/it/SpringDataIT.java]: Yea but "Scyther" is the correct spelling haha

File: src/test/java/com/google/cloud/spanner/r2dbc/it/SpannerR2dbcDialect.java (hunk 3 above)
Comment (dzou) [reply src/test/java/com/google/cloud/spanner/r2dbc/it/SpannerR2dbcDialect.java]: Right, LIMIT appears to be mandatory to use OFFSET: https://cloud.google.com/spanner/docs/query-syntax#limit-clause-and-offset-clause

Turns out something similar is done [MySqlDialect.java](https://github.com/spring-projects/spring-data-jdbc/blob/master/spring-data-relational/src/main/java/org/springframework/data/relational/core/dialect/MySqlDialect.java)

File: src/test/java/com/google/cloud/spanner/r2dbc/it/SpannerR2dbcDialectProvider.java (hunk 4 above)
Comment (dzou) [reply src/test/java/com/google/cloud/spanner/r2dbc/it/SpannerR2dbcDialectProvider.java]: Yeah either that or we could name the metadata constant "spanner" here; either is fine.

File: pom.xml [hunk 5]
```diff
@@ -101,7 +101,12 @@
       <version>1.7.25</version>
       <scope>test</scope>
     </dependency>
-
+    <dependency>
+      <groupId>org.springframework.data</groupId>
+      <artifactId>spring-data-r2dbc</artifactId>
+      <version>1.0.0.BUILD-SNAPSHOT</version>
```
Comment (mp911de) [inline pom.xml]: 1.0M3 is scheduled for end of July.

Comment (eddumelendez) [reply pom.xml]: `spring-data-r2dbc` 1.0 was released https://spring.io/blog/2019/12/06/spring-data-r2dbc-goes-ga

Comment (elefeint) [reply pom.xml]: And now that this is a multi-module project, we can keep the sample in this repo!

**Comment by dzou** [comment]
Closing in favor of #210.

**Pull request closed without merging**